investos/portfolio/risk_model/factor_risk.py

```diff
@@ -1,23 +1,351 @@
 import cvxpy as cvx
+import forecastos as fos  # from pypi
 import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+from scipy.stats import mstats
+from sklearn.preprocessing import StandardScaler
 
 import investos.util as util
 from investos.portfolio.risk_model import BaseRisk
+from investos.portfolio.risk_model.utils.risk_adjustments import (
+    AssetDigonalVarAdjuster,
+    FactorCovAdjuster,
+)
+
+fos_return_uuid_dict = {
+    # For momentum and market beta
+    "return_1d": "ea4d2557-7f8f-476b-b4d3-55917a941bb5",
+}
+fos_risk_factor_uuids_dict = {
+    # Industry
+    "industry": "6f8829e2-6424-4c9a-b05d-d0e5c580859b",
+    # Size
+    "market_cap_open_dil": "dfa7e6a3-671d-41b2-89e3-10b7bdcf7af9",
+    # For liquidity (probably very correlated with size and not needed)
+    "volume": "44fa6e70-dd84-4be4-90a4-cb0f2b62c0ca",  # Daily volume traded, split adjusted, in thousands of shares. ID is FactSet Sym ID.
+    "price_open": "dc41d72a-ba23-4ab2-90ff-6ab776f7a16b",  # Share price at open, split adjusted. ID is FactSet Sym ID.
+    # Leverage
+    "debt_total_prev_1q_to_ebit_ltm": "53a422bf-1dab-4d1e-a9a7-2478a226435b",
+    "debt_total_to_market_cap_open_dil_prev_1q": "d862bc25-e085-442a-acfe-05c880da2d40",  # 'prev_1q debt_total to market_cap_open_dil multiple. ID is FactSet Sym ID.'
+    # Quality
+    "net_income_div_sales_ltm": "7f1e058f-46b6-406f-81a1-d8a5b81371a2",  # ltm profit margin (as % of sales)
```

Comment:
How do you plan on combining quality factors? You have 2.

Reply:
i.e. what data structure implies factors should be blended? I don't see a data structure implying a blend